osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -102,15 +129,27 @@ public abstract class Flashlight : Drawable
 
             public List<BreakPeriod> Breaks = new List<BreakPeriod>();
 
-            private readonly float defaultFlashlightSize;
-            private readonly float sizeMultiplier;
+            private readonly float appliedFlashlightSize;
+            private readonly float changeSizeDecreaseRatio;
             private readonly bool comboBasedSize;
 
+            private readonly int maxChangeSizeTimes;
+            private readonly float changeSizeCombo;
+
             protected Flashlight(ModFlashlight modFlashlight)
             {
-                defaultFlashlightSize = modFlashlight.DefaultFlashlightSize;
-                sizeMultiplier = modFlashlight.SizeMultiplier.Value;
-                comboBasedSize = modFlashlight.ComboBasedSize.Value;
+                changeSizeCombo = modFlashlight.ChangeSizeCombo.Value;
+                maxChangeSizeTimes = modFlashlight.MaxChangeSizeTimes.Value;
+
+                appliedFlashlightSize = modFlashlight.DefaultFlashlightSize * modFlashlight.SizeMultiplier.Value;
+
+                var finalFlashlightSizeBinding = modFlashlight.FinalFlashlightSize;
+                float finalFlashlightSize = finalFlashlightSizeBinding.Value;
```

Comment:
To match the above variable definitions
```suggestion
                float finalFlashlightSize = modFlashlight.FinalFlashlightSize.Value;
```

Reply:
Done.